controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/dns/NameService.java

```diff
@@ -12,13 +12,14 @@
 public interface NameService {
 
     /**
-     * Create a new CNAME record
+     * Create a new record
      *
-     * @param name          The alias to create (lhs of the record)
-     * @param canonicalName The canonical name which the alias should point to (rhs of the record). This must be a FQDN.
+     * @param type The DNS type of record to make, only a small set of types are supported, check with the implementation
+     * @param name Name of the record, e.g. a FQDN for records of type A
+     * @param data Data of the record, e.g. IP address for records of type A
      * @return The created record
      */
-    Record createCname(RecordName name, RecordData canonicalName);
+    Record createRecord(Record.Type type, RecordName name, RecordData data);
```

Comment:
I have considered this, but decided against that because:
* none of the callers of this already have a `Record`, so all call sites would get an additional `new Record(...)`
* OpsDB implementation uses TTL of 1h, while the default value in `Record` is 5m, so either we would have to ignore the TTL in `Record` or all the `new Record(...)` would also have to specify the duration